host/frontend/host-orchestrator/operator.go

```diff
@@ -46,12 +51,26 @@ func main() {
 
 	setupDeviceEndpoint(pool, config, socketPath)
 	r := setupServerRoutes(pool, polledSet, config)
-
 	http.Handle("/", r)
-	log.Println("Client endpoint created")
-	if err := http.ListenAndServe(fmt.Sprint(":", port), nil); err != nil {
-		log.Fatal("ListenAndServe client: ", err)
-	}
+
+	wg := new(sync.WaitGroup)
+	wg.Add(2)
```

Comment:
This makes me nervous: What happens if someone wants to add a new go routine after line 71? They will most likely copy from 65 to 70 and modify as needed, but then they may not change this to add 3 instead.

I know this is the way recommended almost anywhere and there is nothing we can do to change the WaitGroup api, but we can put the call to Add next to the go routines, and do it one by one, so when someone copies they will see the Add() call.

Ideally, I think WaiGroup's Add() should take a closure to be executed in a routine instead of a number and you simply call wait after adding all the closures.

Reply:
great point, changed the way the WaitGroup is incremented in order to avoid what you pointed out, PTAL.